Userland/Libraries/LibWeb/DOM/Range.cpp

```diff
@@ -109,57 +106,38 @@ void Range::update_associated_selection()
     // When the selection is dissociated with its range, associated with a new range, or the
     // associated range's boundary point is mutated either by the user or the content script, the
     // user agent must schedule a selectionchange event on document.
-    schedule_a_selectionchange_event(document);
-
-    // When an input or textarea element provide a text selection and its selection changes (in
-    // either extent or direction), the user agent must schedule a selectionchange event on the
-    // element.
-    if (m_start_container == m_end_container) {
-        if (is<HTML::HTMLInputElement>(*m_start_container))
-            schedule_a_selectionchange_event(verify_cast<HTML::HTMLInputElement>(*m_start_container));
-        if (is<HTML::HTMLTextAreaElement>(*m_start_container))
-            schedule_a_selectionchange_event(verify_cast<HTML::HTMLTextAreaElement>(*m_start_container));
-    }
+    schedule_a_selectionchange_event();
 }
 
 // https://w3c.github.io/selection-api/#scheduling-selectionhange-event
-template<SelectionChangeTarget T>
-void Range::schedule_a_selectionchange_event(T& target)
+void Range::schedule_a_selectionchange_event()
```

Comment:
It's a shame we have to duplicate both these steps and those from "fire a selectionchange event". Maybe we can host them as static functions in another unit?

Reply:
I would definitely consider deduplicating it but steps are basically 1) create event 2) dispatch event. maybe something to revisit in the future if logic for dispatching this event becomes more complicated.

Reply:
ok, you are right. I moved event dispatching into separate file.